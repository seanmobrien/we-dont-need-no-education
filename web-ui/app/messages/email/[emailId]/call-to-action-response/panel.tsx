import React, { useState, useEffect } from 'react';
import { 
  Stack, 
  Paper, 
  Typography, 
  Chip, 
  Box, 
  Grid,
  Divider,
  LinearProgress,
  Accordion,
  AccordionSummary,
  AccordionDetails,
  Link,
  Alert,
  CircularProgress
} from '@mui/material';
import ExpandMoreIcon from '@mui/icons-material/ExpandMore';
import OpenInNewIcon from '@mui/icons-material/OpenInNew';
import { CallToActionResponseDetails, CallToActionDetails } from '@/data-models/api';
import { getCallToAction } from '@/lib/api/email/properties/client';
import { useParams } from 'next/navigation';

const formatDate = (date: Date | null): string => {
  if (!date) return 'Not set';
  return new Intl.DateTimeFormat('en-US', {
    year: 'numeric',
    month: 'short', 
    day: 'numeric',
    hour: '2-digit',
    minute: '2-digit'
  }).format(date);
};

const formatPercentage = (value: number | null): string => {
  if (value === null || value === undefined) return 'N/A';
  return `${Math.round(value)}%`;
};

const formatScore = (value: number | null): string => {
  if (value === null || value === undefined) return 'N/A';
  return value.toFixed(2);
};

export const ResponsiveActionPanel = ({ row }: { row: CallToActionResponseDetails }) => {
  const { emailId } = useParams();
  const [relatedCTA, setRelatedCTA] = useState<CallToActionDetails | null>(null);
  const [loading, setLoading] = useState(true);
  const [error, setError] = useState<string | null>(null);

  useEffect(() => {
    const fetchRelatedCTA = async () => {
      try {
        setLoading(true);
        const result = await getCallToAction({ 
          emailId: emailId as string,
          page: 1,
          num: 100 // Get all CTAs
        });
        
        // Find the CTA that this response is related to
        const relatedCta = result.results.find(cta => 
          cta.propertyId === row.actionPropertyId
        );
        setRelatedCTA(relatedCta || null);
      } catch (err) {
        setError('Failed to load related call-to-action');
        console.error('Error fetching related CTA:', err);
      } finally {
        setLoading(false);
      }
    };

    if (row.actionPropertyId && emailId) {
      fetchRelatedCTA();
    }
  }, [row.actionPropertyId, emailId]);

  return (
    <Stack sx={{ py: 2, height: '100%', boxSizing: 'border-box' }} direction="column">
      <Paper sx={{ flex: 1, mx: 'auto', width: '95%', p: 2 }}>
        <Stack direction="column" spacing={2}>
          {/* Header */}
          <Typography variant="h5" component="h2" gutterBottom>
            Responsive Action Details
          </Typography>
          
          {/* Description */}
          <Box>
            <Typography variant="h6" gutterBottom>Response Description</Typography>
            <Typography variant="body1" sx={{ backgroundColor: 'grey.100', p: 2, borderRadius: 1 }}>
              {row.value}
            </Typography>
          </Box>

          {/* Progress and Timing */}
          <Grid container spacing={2}>
            <Grid item xs={12} md={6}>
              <Typography variant="h6" gutterBottom>Completion Progress</Typography>
              <Box sx={{ mb: 1 }}>
                <Typography variant="body2" color="textSecondary">
                  {formatPercentage(row.completionPercentage)} Complete
                </Typography>
              </Box>
              <LinearProgress 
                variant="determinate" 
                value={row.completionPercentage || 0} 
                sx={{ height: 8, borderRadius: 1 }}
              />
            </Grid>
            <Grid item xs={12} md={6}>
              <Typography variant="h6" gutterBottom>Response Timing</Typography>
              <Typography variant="body1" fontWeight="bold">
                {formatDate(row.responseTimestamp)}
              </Typography>
            </Grid>
          </Grid>

          {/* Status Indicators */}
          <Box>
            <Typography variant="h6" gutterBottom>Status</Typography>
            <Stack direction="row" spacing={1} flexWrap="wrap">
              {row.inferred !== undefined && (
                <Chip 
                  label={row.inferred ? "Inferred" : "Direct"} 
                  color={row.inferred ? "secondary" : "primary"}
                  size="small"
                />
              )}
            </Stack>
          </Box>

          {/* Scores and Ratings */}
          <Grid container spacing={2}>
            <Grid item xs={12} md={4}>
              <Typography variant="subtitle1" fontWeight="bold">Severity Score</Typography>
              <Typography variant="body2">{formatScore(row.severity)}</Typography>
            </Grid>
            <Grid item xs={12} md={4}>
              <Typography variant="subtitle1" fontWeight="bold">Sentiment Score</Typography>
              <Typography variant="body2">{formatScore(row.sentiment)}</Typography>
            </Grid>
            <Grid item xs={12} md={4}>
              <Typography variant="subtitle1" fontWeight="bold">Chapter 13 Compliance</Typography>
              <Typography variant="body2">{formatScore(row.compliance_average_chapter_13)}</Typography>
            </Grid>
          </Grid>

          <Divider />

          {/* Expandable Sections */}
          <Stack spacing={1}>
            {/* Reasons and Analysis */}
            {(row.severity_reasons?.length || 
              row.sentiment_reasons?.length || 
              row.compliance_chapter_13_reasons?.length) && (
              <Accordion>
                <AccordionSummary expandIcon={<ExpandMoreIcon />}>
                  <Typography variant="h6">Analysis and Reasoning</Typography>
                </AccordionSummary>
                <AccordionDetails>
                  <Stack spacing={2}>
                    {row.severity_reasons?.length && (
                      <Box>
                        <Typography variant="subtitle1" fontWeight="bold">Severity Reasons</Typography>
                        <Stack direction="row" spacing={1} flexWrap="wrap">
                          {row.severity_reasons.map((reason, index) => (
                            <Chip key={index} label={reason} variant="outlined" size="small" />
                          ))}
                        </Stack>
                      </Box>
                    )}
                    {row.sentiment_reasons?.length && (
                      <Box>
                        <Typography variant="subtitle1" fontWeight="bold">Sentiment Reasons</Typography>
                        <Stack direction="row" spacing={1} flexWrap="wrap">
                          {row.sentiment_reasons.map((reason, index) => (
                            <Chip key={index} label={reason} variant="outlined" size="small" />
                          ))}
                        </Stack>
                      </Box>
                    )}
                    {row.compliance_chapter_13_reasons?.length && (
                      <Box>
                        <Typography variant="subtitle1" fontWeight="bold">Chapter 13 Compliance Reasons</Typography>
                        <Stack direction="row" spacing={1} flexWrap="wrap">
                          {row.compliance_chapter_13_reasons.map((reason, index) => (
                            <Chip key={index} label={reason} variant="outlined" size="small" />
                          ))}
                        </Stack>
                      </Box>
                    )}
                  </Stack>
                </AccordionDetails>
              </Accordion>
            )}

            {/* Related Call-to-Action */}
            <Accordion>
              <AccordionSummary expandIcon={<ExpandMoreIcon />}>
                <Typography variant="h6">
                  Related Call-to-Action
                </Typography>
              </AccordionSummary>
              <AccordionDetails>
                {loading ? (
                  <Box display="flex" justifyContent="center" p={2}>
                    <CircularProgress size={24} />
                  </Box>
                ) : error ? (
                  <Alert severity="error">{error}</Alert>
                ) : !relatedCTA ? (
                  <Typography variant="body2" color="textSecondary">
                    No related call-to-action found.
                  </Typography>
                ) : (
                  <Paper variant="outlined" sx={{ p: 2 }}>
                    <Stack direction="row" justifyContent="space-between" alignItems="flex-start">
                      <Box flex={1}>
                        <Typography variant="subtitle1" fontWeight="bold" gutterBottom>
                          Call-to-Action Details
                        </Typography>
                        <Typography variant="body2" sx={{ mb: 1 }}>
                          {relatedCTA.value}
                        </Typography>
                        <Grid container spacing={2}>
                          <Grid item xs={6}>
                            <Typography variant="caption" color="textSecondary">
                              Completion: {formatPercentage(relatedCTA.completion_percentage)}
                            </Typography>
                          </Grid>
                          <Grid item xs={6}>
                            <Typography variant="caption" color="textSecondary">
                              Status: {relatedCTA.inferred ? 'Inferred' : 'Direct'}
                            </Typography>
                          </Grid>
                          <Grid item xs={6}>
                            <Typography variant="caption" color="textSecondary">
                              Opened: {formatDate(relatedCTA.opened_date)}
                            </Typography>
                          </Grid>
                          <Grid item xs={6}>
                            <Typography variant="caption" color="textSecondary">
                              Due: {formatDate(relatedCTA.compliancy_close_date)}
                            </Typography>
                          </Grid>
                        </Grid>
                      </Box>
                      <Link
                        href={`/messages/email/${emailId}/call-to-action?highlight=${relatedCTA.propertyId}`}
                        target="_blank"
                        rel="noopener noreferrer"
                        sx={{ ml: 2 }}
                      >
                        <OpenInNewIcon fontSize="small" />
                      </Link>
                    </Stack>
                  </Paper>
                )}
              </AccordionDetails>
            </Accordion>

            {/* Policy Basis and Tags */}
            <Accordion>
              <AccordionSummary expandIcon={<ExpandMoreIcon />}>
                <Typography variant="h6">Metadata</Typography>
              </AccordionSummary>
              <AccordionDetails>
                <Stack spacing={2}>
                  <Box>
                    <Typography variant="subtitle1" fontWeight="bold" gutterBottom>
                      Policy Basis
                    </Typography>
                    {row.policy_basis?.length ? (
                      <Stack direction="row" spacing={1} flexWrap="wrap">
                        {row.policy_basis.map((item, index) => (
                          <Chip key={index} label={item} color="primary" variant="outlined" />
                        ))}
                      </Stack>
                    ) : (
                      <Typography variant="body2" color="textSecondary">
                        No policy basis specified
                      </Typography>
                    )}
                  </Box>
                  <Box>
                    <Typography variant="subtitle1" fontWeight="bold" gutterBottom>
                      Tags
                    </Typography>
                    {row.tags?.length ? (
                      <Stack direction="row" spacing={1} flexWrap="wrap">
                        {row.tags.map((item, index) => (
                          <Chip key={index} label={item} color="secondary" variant="outlined" />
                        ))}
                      </Stack>
                    ) : (
                      <Typography variant="body2" color="textSecondary">
                        No tags specified
                      </Typography>
                    )}
                  </Box>
                </Stack>
              </AccordionDetails>
            </Accordion>
          </Stack>
        </Stack>
      </Paper>
    </Stack>
  );
};

export default ResponsiveActionPanel;
