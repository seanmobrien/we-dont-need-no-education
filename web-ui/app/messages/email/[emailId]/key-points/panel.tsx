import { EmailMasterPanel } from '@/components/mui/data-grid';
import { KeyPointsDetails } from '@/data-models/api';

export const KeyPointsPanel = ({
  row: keyPoint,
}: {
  row: KeyPointsDetails;
}) => {
  return <EmailMasterPanel title="Key Point" row={keyPoint} />;
};
