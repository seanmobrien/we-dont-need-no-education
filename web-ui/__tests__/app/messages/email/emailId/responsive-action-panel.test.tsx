/* eslint-disable @typescript-eslint/no-explicit-any */
import React from 'react';
import { render, screen, waitFor } from '@testing-library/react';
import '@testing-library/jest-dom';
import { ResponsiveActionPanel } from '@/app/messages/email/[emailId]/call-to-action-response/panel';
import { CallToActionResponseDetails } from '@/data-models/api';
import * as client from '@/lib/api/email/properties/client';

// Mock the API client
jest.mock('@/lib/api/email/properties/client');
const mockedClient = client as jest.Mocked<typeof client>;

// Mock next/navigation
jest.mock('next/navigation', () => ({
  useParams: () => ({ emailId: 'test-email-id' }),
}));

// Mock MUI components that may cause theme issues
jest.mock(
  '@mui/material/LinearProgress',
  () =>
    ({ value, ...props }: { value?: number }) => (
      <div data-testid="linear-progress" data-value={value} {...props} />
    ),
);

jest.mock('@mui/material/CircularProgress', () => () => (
  <div data-testid="circular-progress" />
));

const mockResponseDetails: CallToActionResponseDetails = {
  propertyId: 'response-test-id',
  documentId: 1,
  createdOn: new Date('2023-01-05'),
  actionPropertyId: 'cta-test-id',
  completionPercentage: 80,
  responseTimestamp: new Date('2023-01-05T10:30:00'),
  value: 'Detailed response to the call to action',
  policy_basis: ['FERPA', 'COPPA'],
  tags: ['response', 'completed'],
  severity: 0.6,
  severity_reasons: ['Moderate severity'],
  inferred: false,
  sentiment: 0.7,
  sentiment_reasons: ['Positive response'],
  compliance_average_chapter_13: 0.85,
  compliance_chapter_13_reasons: ['Meets Chapter 13 requirements'],
};

const mockRelatedCTA = {
  propertyId: 'cta-test-id',
  documentId: 1,
  createdOn: new Date('2023-01-01'),
  value: 'Original call to action that this responds to',
  completion_percentage: 75,
  inferred: false,
  opened_date: new Date('2023-01-01'),
  compliancy_close_date: new Date('2023-02-01'),
  closed_date: null,
  compliance_date_enforceable: true,
};

describe('ResponsiveActionPanel', () => {
  beforeEach(() => {
    jest.clearAllMocks();
    mockedClient.getCallToAction.mockResolvedValue({
      results: [mockRelatedCTA],
      pageStats: { page: 1, num: 10, total: 1 },
    });
  });

  it('renders responsive action details correctly', async () => {
    render(<ResponsiveActionPanel row={mockResponseDetails} />);

    expect(
      screen.getByText('Responsive Action (response-test-id)'),
    ).toBeInTheDocument();
    expect(
      screen.getByText('Detailed response to the call to action'),
    ).toBeInTheDocument();
    expect(screen.getByText('80% Complete')).toBeInTheDocument();
  });

  it('displays progress bar with correct value', async () => {
    render(<ResponsiveActionPanel row={mockResponseDetails} />);

    const progressBar = screen.getByTestId('linear-progress');
    expect(progressBar).toHaveAttribute('data-value', '80');
  });

  it('shows response timestamp correctly', async () => {
    render(<ResponsiveActionPanel row={mockResponseDetails} />);

    expect(screen.getByText(/Jan 5, 2023/)).toBeInTheDocument();
  });

  it('displays status chips correctly', async () => {
    render(<ResponsiveActionPanel row={mockResponseDetails} />);

    expect(screen.getByText('Direct')).toBeInTheDocument(); // since inferred is false
  });

  it('shows all scores correctly', async () => {
    render(<ResponsiveActionPanel row={mockResponseDetails} />);

    expect(screen.getByText('0.60')).toBeInTheDocument(); // severity
    expect(screen.getByText('0.70')).toBeInTheDocument(); // sentiment
    expect(screen.getByText('0.85')).toBeInTheDocument(); // compliance_average_chapter_13
  });

  it('displays policy basis and tags in metadata', async () => {
    render(<ResponsiveActionPanel row={mockResponseDetails} />);

    expect(screen.getByText('FERPA')).toBeInTheDocument();
    expect(screen.getByText('COPPA')).toBeInTheDocument();
    expect(screen.getByText('response')).toBeInTheDocument();
    expect(screen.getByText('completed')).toBeInTheDocument();
  });

  it('fetches and displays related call-to-action', async () => {
    render(<ResponsiveActionPanel row={mockResponseDetails} />);

    await waitFor(() => {
      expect(mockedClient.getCallToAction).toHaveBeenCalledWith({
        emailId: 'test-email-id',
        page: 1,
        num: 100,
      });
    });

    await waitFor(() => {
      expect(screen.getByText('Call-to-Action Details')).toBeInTheDocument();
      expect(
        screen.getByText('Original call to action that this responds to'),
      ).toBeInTheDocument();
    });
  });

  it('shows loading state while fetching related CTA', async () => {
    // Make the API call take some time
    mockedClient.getCallToAction.mockImplementation(
      () =>
        new Promise((resolve) =>
          setTimeout(
            () =>
              resolve({
                results: [mockRelatedCTA],
                pageStats: { page: 1, num: 10, total: 1 },
              }),
            100,
          ),
        ) as any,
    );

    render(<ResponsiveActionPanel row={mockResponseDetails} />);

    expect(screen.getByTestId('circular-progress')).toBeInTheDocument();
  });

  it('handles API error gracefully', async () => {
    mockedClient.getCallToAction.mockRejectedValue(new Error('API Error'));

    render(<ResponsiveActionPanel row={mockResponseDetails} />);

    await waitFor(() => {
      expect(
        screen.getByText('Failed to load related call-to-action'),
      ).toBeInTheDocument();
    });
  });

  it('handles no related CTA found', async () => {
    mockedClient.getCallToAction.mockResolvedValue({
      results: [],
      pageStats: { page: 1, num: 10, total: 0 },
    });

    render(<ResponsiveActionPanel row={mockResponseDetails} />);

    await waitFor(() => {
      expect(
        screen.getByText('No related call-to-action found.'),
      ).toBeInTheDocument();
    });
  });

  it('displays reasoning sections when available', async () => {
    render(<ResponsiveActionPanel row={mockResponseDetails} />);

    expect(screen.getByText('Analysis and Reasoning')).toBeInTheDocument();
    expect(screen.getByText('Moderate severity')).toBeInTheDocument();
    expect(screen.getByText('Positive response')).toBeInTheDocument();
    expect(
      screen.getByText('Meets Chapter 13 requirements'),
    ).toBeInTheDocument();
  });

  it('handles missing optional fields gracefully', async () => {
    const minimalResponse: CallToActionResponseDetails = {
      propertyId: 'response-minimal',
      documentId: 1,
      createdOn: new Date('2023-01-05'),
      actionPropertyId: 'cta-test-id',
      completionPercentage: 50,
      responseTimestamp: new Date('2023-01-05'),
      value: 'Minimal response',
    };

    render(<ResponsiveActionPanel row={minimalResponse} />);

    expect(screen.getByText('Minimal response')).toBeInTheDocument();
    expect(screen.getByText('50% Complete')).toBeInTheDocument();
  });
});
