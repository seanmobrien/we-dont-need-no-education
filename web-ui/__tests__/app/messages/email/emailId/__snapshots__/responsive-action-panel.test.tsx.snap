// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResponsiveActionPanel handles API error gracefully 1`] = `
<div
  class="MuiStack-root css-1kq55ho-MuiStack-root"
>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-avbnl1-MuiPaper-root"
    style="--Paper-shadow: 0px 2px 1px -1px rgba(0,0,0,0.2),0px 1px 1px 0px rgba(0,0,0,0.14),0px 1px 3px 0px rgba(0,0,0,0.12);"
  >
    <div
      class="MuiStack-root css-1a7f82c-MuiStack-root"
    >
      <h6
        class="MuiTypography-root MuiTypography-h6 css-1miy0lu-MuiTypography-root"
      >
        Responsive Action
         
        (response-test-id)
      </h6>
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation2 css-1yruui0-MuiPaper-root"
        style="--Paper-shadow: 0px 3px 1px -2px rgba(0,0,0,0.2),0px 2px 2px 0px rgba(0,0,0,0.14),0px 1px 5px 0px rgba(0,0,0,0.12);"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 MuiTypography-alignLeft css-10gu7ti-MuiTypography-root"
          style="--Typography-textAlign: left;"
        >
          Detailed response to the call to action
        </p>
      </div>
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-row MuiGrid-spacing-xs-2 css-zjhuaj-MuiGrid-root"
      >
        <div
          class="MuiGrid-root MuiGrid-direction-xs-row css-1gb4p1x-MuiGrid-root"
        >
          <h6
            class="MuiTypography-root MuiTypography-h6 MuiTypography-gutterBottom css-1wmifpn-MuiTypography-root"
          >
            Completion Progress
          </h6>
          <div
            class="MuiBox-root css-19idom"
          >
            <p
              class="MuiTypography-root MuiTypography-body2 css-1a1whku-MuiTypography-root"
            >
              80%
               Complete
            </p>
          </div>
          <span
            aria-valuemax="100"
            aria-valuemin="0"
            aria-valuenow="80"
            class="MuiLinearProgress-root MuiLinearProgress-colorPrimary MuiLinearProgress-determinate css-13s5bod-MuiLinearProgress-root"
            role="progressbar"
          >
            <span
              class="MuiLinearProgress-bar MuiLinearProgress-bar1 MuiLinearProgress-barColorPrimary MuiLinearProgress-bar1Determinate css-l16vtb-MuiLinearProgress-bar1"
              style="transform: translateX(-20%);"
            />
          </span>
        </div>
        <div
          class="MuiGrid-root MuiGrid-direction-xs-row css-1gb4p1x-MuiGrid-root"
        >
          <h6
            class="MuiTypography-root MuiTypography-h6 MuiTypography-gutterBottom css-1wmifpn-MuiTypography-root"
          >
            Response Timing
          </h6>
          <p
            class="MuiTypography-root MuiTypography-body1 css-lhsbas-MuiTypography-root"
          >
            Jan 5, 2023, 10:30 AM
          </p>
        </div>
      </div>
      <div
        class="MuiBox-root css-0"
      >
        <h6
          class="MuiTypography-root MuiTypography-h6 MuiTypography-gutterBottom css-1wmifpn-MuiTypography-root"
        >
          Status
        </h6>
        <div
          class="MuiStack-root css-ofxwhm-MuiStack-root"
        >
          <div
            class="MuiChip-root MuiChip-filled MuiChip-sizeSmall MuiChip-colorPrimary MuiChip-filledPrimary css-1abuvwb-MuiChip-root"
          >
            <span
              class="MuiChip-label MuiChip-labelSmall css-eccknh-MuiChip-label"
            >
              Direct
            </span>
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-row MuiGrid-spacing-xs-2 css-zjhuaj-MuiGrid-root"
      >
        <div
          class="MuiGrid-root MuiGrid-direction-xs-row css-u42tu4-MuiGrid-root"
        >
          <h6
            class="MuiTypography-root MuiTypography-subtitle1 css-or3kzz-MuiTypography-root"
          >
            Severity Score
          </h6>
          <p
            class="MuiTypography-root MuiTypography-body2 css-7ctihp-MuiTypography-root"
          >
            0.60
          </p>
        </div>
        <div
          class="MuiGrid-root MuiGrid-direction-xs-row css-u42tu4-MuiGrid-root"
        >
          <h6
            class="MuiTypography-root MuiTypography-subtitle1 css-or3kzz-MuiTypography-root"
          >
            Sentiment Score
          </h6>
          <p
            class="MuiTypography-root MuiTypography-body2 css-7ctihp-MuiTypography-root"
          >
            0.70
          </p>
        </div>
        <div
          class="MuiGrid-root MuiGrid-direction-xs-row css-u42tu4-MuiGrid-root"
        >
          <h6
            class="MuiTypography-root MuiTypography-subtitle1 css-or3kzz-MuiTypography-root"
          >
            Chapter 13 Compliance
          </h6>
          <p
            class="MuiTypography-root MuiTypography-body2 css-7ctihp-MuiTypography-root"
          >
            0.85
          </p>
        </div>
      </div>
      <hr
        class="MuiDivider-root MuiDivider-fullWidth css-1jpc804-MuiDivider-root"
      />
      <div
        class="MuiStack-root css-jfdv4h-MuiStack-root"
      >
        <div
          class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiAccordion-root MuiAccordion-rounded MuiAccordion-gutters css-1808mag-MuiPaper-root-MuiAccordion-root"
          style="--Paper-shadow: 0px 2px 1px -1px rgba(0,0,0,0.2),0px 1px 1px 0px rgba(0,0,0,0.14),0px 1px 3px 0px rgba(0,0,0,0.12);"
        >
          <h3
            class="MuiAccordion-heading css-cy7rkm-MuiAccordion-heading"
          >
            <button
              aria-expanded="false"
              class="MuiButtonBase-root MuiAccordionSummary-root MuiAccordionSummary-gutters css-1f1pi2l-MuiButtonBase-root-MuiAccordionSummary-root"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiAccordionSummary-content MuiAccordionSummary-contentGutters css-yfrx4k-MuiAccordionSummary-content"
              >
                <h6
                  class="MuiTypography-root MuiTypography-h6 css-1miy0lu-MuiTypography-root"
                >
                  Analysis and Reasoning
                </h6>
              </span>
              <span
                class="MuiAccordionSummary-expandIconWrapper css-1wqf3nl-MuiAccordionSummary-expandIconWrapper"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
                  data-testid="ExpandMoreIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M16.59 8.59 12 13.17 7.41 8.59 6 10l6 6 6-6z"
                  />
                </svg>
              </span>
            </button>
          </h3>
          <div
            class="MuiCollapse-root MuiCollapse-vertical MuiCollapse-hidden css-cwrbtg-MuiCollapse-root"
            style="min-height: 0px;"
          >
            <div
              class="MuiCollapse-wrapper MuiCollapse-vertical css-1x6hinx-MuiCollapse-wrapper"
            >
              <div
                class="MuiCollapse-wrapperInner MuiCollapse-vertical css-1i4ywhz-MuiCollapse-wrapperInner"
              >
                <div
                  class="MuiAccordion-region"
                  role="region"
                >
                  <div
                    class="MuiAccordionDetails-root css-1lj39kh-MuiAccordionDetails-root"
                  >
                    <div
                      class="MuiStack-root css-1sazv7p-MuiStack-root"
                    >
                      <div
                        class="MuiBox-root css-0"
                      >
                        <h6
                          class="MuiTypography-root MuiTypography-subtitle1 css-or3kzz-MuiTypography-root"
                        >
                          Severity Reasons
                        </h6>
                        <div
                          class="MuiStack-root css-ofxwhm-MuiStack-root"
                        >
                          <div
                            class="MuiChip-root MuiChip-outlined MuiChip-sizeSmall MuiChip-colorDefault MuiChip-outlinedDefault css-1d2be57-MuiChip-root"
                          >
                            <span
                              class="MuiChip-label MuiChip-labelSmall css-19m61dl-MuiChip-label"
                            >
                              Moderate severity
                            </span>
                          </div>
                        </div>
                      </div>
                      <div
                        class="MuiBox-root css-0"
                      >
                        <h6
                          class="MuiTypography-root MuiTypography-subtitle1 css-or3kzz-MuiTypography-root"
                        >
                          Sentiment Reasons
                        </h6>
                        <div
                          class="MuiStack-root css-ofxwhm-MuiStack-root"
                        >
                          <div
                            class="MuiChip-root MuiChip-outlined MuiChip-sizeSmall MuiChip-colorDefault MuiChip-outlinedDefault css-1d2be57-MuiChip-root"
                          >
                            <span
                              class="MuiChip-label MuiChip-labelSmall css-19m61dl-MuiChip-label"
                            >
                              Positive response
                            </span>
                          </div>
                        </div>
                      </div>
                      <div
                        class="MuiBox-root css-0"
                      >
                        <h6
                          class="MuiTypography-root MuiTypography-subtitle1 css-or3kzz-MuiTypography-root"
                        >
                          Chapter 13 Compliance Reasons
                        </h6>
                        <div
                          class="MuiStack-root css-ofxwhm-MuiStack-root"
                        >
                          <div
                            class="MuiChip-root MuiChip-outlined MuiChip-sizeSmall MuiChip-colorDefault MuiChip-outlinedDefault css-1d2be57-MuiChip-root"
                          >
                            <span
                              class="MuiChip-label MuiChip-labelSmall css-19m61dl-MuiChip-label"
                            >
                              Meets Chapter 13 requirements
                            </span>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiAccordion-root MuiAccordion-rounded MuiAccordion-gutters css-1808mag-MuiPaper-root-MuiAccordion-root"
          style="--Paper-shadow: 0px 2px 1px -1px rgba(0,0,0,0.2),0px 1px 1px 0px rgba(0,0,0,0.14),0px 1px 3px 0px rgba(0,0,0,0.12);"
        >
          <h3
            class="MuiAccordion-heading css-cy7rkm-MuiAccordion-heading"
          >
            <button
              aria-expanded="false"
              class="MuiButtonBase-root MuiAccordionSummary-root MuiAccordionSummary-gutters css-1f1pi2l-MuiButtonBase-root-MuiAccordionSummary-root"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiAccordionSummary-content MuiAccordionSummary-contentGutters css-yfrx4k-MuiAccordionSummary-content"
              >
                <h6
                  class="MuiTypography-root MuiTypography-h6 css-1miy0lu-MuiTypography-root"
                >
                  Related Call-to-Action
                </h6>
              </span>
              <span
                class="MuiAccordionSummary-expandIconWrapper css-1wqf3nl-MuiAccordionSummary-expandIconWrapper"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
                  data-testid="ExpandMoreIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M16.59 8.59 12 13.17 7.41 8.59 6 10l6 6 6-6z"
                  />
                </svg>
              </span>
            </button>
          </h3>
          <div
            class="MuiCollapse-root MuiCollapse-vertical MuiCollapse-hidden css-cwrbtg-MuiCollapse-root"
            style="min-height: 0px;"
          >
            <div
              class="MuiCollapse-wrapper MuiCollapse-vertical css-1x6hinx-MuiCollapse-wrapper"
            >
              <div
                class="MuiCollapse-wrapperInner MuiCollapse-vertical css-1i4ywhz-MuiCollapse-wrapperInner"
              >
                <div
                  class="MuiAccordion-region"
                  role="region"
                >
                  <div
                    class="MuiAccordionDetails-root css-1lj39kh-MuiAccordionDetails-root"
                  >
                    <div
                      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation0 MuiAlert-root MuiAlert-colorError MuiAlert-standardError MuiAlert-standard css-3lhuho-MuiPaper-root-MuiAlert-root"
                      role="alert"
                      style="--Paper-shadow: none;"
                    >
                      <div
                        class="MuiAlert-icon css-vab54s-MuiAlert-icon"
                      >
                        <svg
                          aria-hidden="true"
                          class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1ckov0h-MuiSvgIcon-root"
                          data-testid="ErrorOutlineIcon"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M11 15h2v2h-2zm0-8h2v6h-2zm.99-5C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zM12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                          />
                        </svg>
                      </div>
                      <div
                        class="MuiAlert-message css-zioonp-MuiAlert-message"
                      >
                        Failed to load related call-to-action
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <section
        class="MuiBox-root css-cxvo6t"
      >
        <p
          class="MuiTypography-root MuiTypography-body2 MuiTypography-alignLeft css-xlaprp-MuiTypography-root"
          style="--Typography-textAlign: left;"
        >
          Policy Basis
        </p>
        <div
          class="MuiTypography-root MuiTypography-body1 MuiTypography-alignLeft css-1j3qii7-MuiTypography-root"
          style="--Typography-textAlign: left;"
        >
          <div
            class="MuiChip-root MuiChip-filled MuiChip-sizeMedium MuiChip-colorDefault MuiChip-filledDefault css-5c51lm-MuiChip-root"
            style="margin: 2px;"
          >
            <span
              class="MuiChip-label MuiChip-labelMedium css-1dybbl5-MuiChip-label"
            >
              FERPA
            </span>
          </div>
          <div
            class="MuiChip-root MuiChip-filled MuiChip-sizeMedium MuiChip-colorDefault MuiChip-filledDefault css-5c51lm-MuiChip-root"
            style="margin: 2px;"
          >
            <span
              class="MuiChip-label MuiChip-labelMedium css-1dybbl5-MuiChip-label"
            >
              COPPA
            </span>
          </div>
        </div>
        <p
          class="MuiTypography-root MuiTypography-body2 MuiTypography-alignLeft css-1ctvvts-MuiTypography-root"
          style="--Typography-textAlign: left;"
        >
          Tags
        </p>
        <div
          class="MuiTypography-root MuiTypography-body1 MuiTypography-alignLeft css-1j3qii7-MuiTypography-root"
          style="--Typography-textAlign: left;"
        >
          <div
            class="MuiChip-root MuiChip-filled MuiChip-sizeMedium MuiChip-colorDefault MuiChip-filledDefault css-5c51lm-MuiChip-root"
            style="margin: 2px;"
          >
            <span
              class="MuiChip-label MuiChip-labelMedium css-1dybbl5-MuiChip-label"
            >
              response
            </span>
          </div>
          <div
            class="MuiChip-root MuiChip-filled MuiChip-sizeMedium MuiChip-colorDefault MuiChip-filledDefault css-5c51lm-MuiChip-root"
            style="margin: 2px;"
          >
            <span
              class="MuiChip-label MuiChip-labelMedium css-1dybbl5-MuiChip-label"
            >
              completed
            </span>
          </div>
        </div>
      </section>
    </div>
  </div>
</div>
`;
